deid/dicom/pixels/clean.py

```diff
@@ -153,36 +153,34 @@ def clean(self, fix_interpretation=True, pixel_data_attribute="PixelData"):
                     # Each is a list with [value, coordinate]
                     mask_value, new_coordinates = coordinate_set
 
-                    for new_coordinate in new_coordinates:
-
-                        # Case 1: an "all" indicates applying to entire image
-                        if new_coordinate.lower() == "all":
-
-                            # no frames, just X, Y
-                            if len(self.original.shape) == 2:
-                                # minr, minc, maxr, maxc = [0, 0, Y, X]
-                                new_coordinate = [
-                                    0,
-                                    0,
-                                    self.original.shape[1],
-                                    self.original.shape[0],
-                                ]
-
-                            # (frames, X, Y, channel) OR (frames, X,Y)
-                            if len(self.original.shape) >= 3:
-                                new_coordinate = [
-                                    0,
-                                    0,
-                                    self.original.shape[2],
-                                    self.original.shape[1],
-                                ]
-
+                    # Case 1: an "all" indicates applying to entire image
+                    if new_coordinates.lower() == "all":
+
+                        # no frames, just X, Y
+                        if len(self.original.shape) == 2:
+                            # minr, minc, maxr, maxc = [0, 0, Y, X]
+                            new_coordinates = [
```

Comment:
And then this is the singular of that, new_coordinate (setting new_coordinates here would overwrite the original variable. The reason that we allow a rule to have more than one coordinate is because there might be more than one relevant region.

Reply:
Absolutely. This is intentional - see my comment above.  With this overwriting we are still allowing for multiple coordinates - when specified as in the deid.dicom recipe.